src/SignalR/server/SignalR/src/HubEndpointRouteBuilderExtensions.cs

```diff
@@ -46,6 +46,16 @@ public static class HubEndpointRouteBuilderExtensions
                                                 "'IServiceCollection.AddSignalR' inside the call to 'ConfigureServices(...)' in the application startup code.");
         }
 
+        using var scope1 = endpoints.ServiceProvider.CreateScope();
+        var hub1 = scope1.ServiceProvider.GetService<THub>();
+        using var scope2 = endpoints.ServiceProvider.CreateScope();
+        var hub2 = scope2.ServiceProvider.GetService<THub>();
+        if (hub1 is not null && hub2 is not null &&
+            hub1 == hub2)
```

Comment:
If we do this, we should probably check for reference equality. This also has the side effect of eagerly initializing all the hubs dependencies. I'm not sure how we feel about that.

Reply:
> This also has the side effect of eagerly initializing all the hubs dependencies. I'm not sure how we feel about that

Yeah, I'm not sure how I feel about it either :D